app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -0,0 +1,345 @@
+package com.github.se.cyrcle.ui.gambling
+
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.viewinterop.AndroidView
+import kotlin.math.cos
+import kotlin.math.sin
+
+class WheelView
+@JvmOverloads
+constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
+    View(context, attrs, defStyleAttr) {
+
+  private val TAG = "WheelView"
+
+  private val paint = Paint(Paint.ANTI_ALIAS_FLAG)
+  private val segments =
+      listOf(
+          Segment("Nothing", Color.GRAY),
+          Segment("Legendary", Color.rgb(255, 215, 0)),
+          Segment("Common", Color.GREEN),
+          Segment("Rare", Color.BLUE),
+          Segment("Epic", Color.rgb(128, 0, 128)))
+
+  private val probabilities =
```

Comment:
The probabilities could be grouped within the Segment dataclass to avoid confusing indices between the two lists